client/sites/components/sites-dataviews/actions.tsx

```diff
@@ -201,7 +201,7 @@ export const isActionEligible = (
 			return ( site: SiteExcerptData ) => {
 				const canManageOptions = capabilities[ site.ID ]?.manage_options;
 				if (
-					! canManageOptions ||
+					( ! canManageOptions && ! site?.is_deleted ) ||
```

Comment:
If site is not deleted then we shouldn't show the `restore` menu item anyway. I don't think we should interfere the site status with the `canManageOptions`.

Seems like the `capabilities` and even the site status are not refreshed in Redux?

Reply:
> If site is not deleted then we shouldn't show the restore menu item anyway. I don't think we should interfere the site status with the canManageOptions.

That's a very good point, basically we can remove the whole `canManageOptions from here and delegate the condition to the [following return](https://github.com/Automattic/wp-calypso/blob/6f32b3d7f22bc877c0697f07f3e26e41429b902d/client/sites/components/sites-dataviews/actions.tsx#L212). I will do the changes.

> Seems like the capabilities and even the site status are not refreshed in Redux?

I didn't realize of that. What are the implications of this? Do you think I should do further investigations as part of this task?

Reply:
Perhaps I wasn't clear enough. `canManageOptions` should be required. We don't want any user other than the admin restoring the site.

From the reports, I understand that this happens only after an account is restored. That's why I **assume** that user permissions are not yet refreshed in Redux.

Reply:
Ok, just to double check, in this scenario `canManageOptions` is undefined after clearing the local state. So that should be investigated further as that (`canMangeOptions` being undefined) can be considered the real problem, right?

Reply:
Regarding other user than the admin restoring the site, I have tested adding other user (as `editor`), and the site is not listed after deleting it


https://github.com/user-attachments/assets/30ef4885-5ea9-40c4-a647-302d45a6d6c6

Reply:
@cpapazoglou, after looking at the code, I have found that deleted sites are only available to admins fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Szl%2Qoybtf%2Qshapgvbaf.cuc%3Se%3Q5q9oqpq2%2387-og so this code will not apply to non-admins. 

On the other hand, the `manage_options` is not returned for deleted sites so admins cannot restore a site (in some scenarios where the Redux state is cleared). Do you think we should try to return `manage_options` from the backend in this scenario instead of updating this condition?

Reply:
I assume that we have used a guard both in the frontend and the backend. We can either:

- rely on the backend guard only ( adding a comment about this in the frontend )
- exploring why `capabilities` are not refreshed

I think I am fine with the first approach. Testing it now.

Reply:
To add more context, I have found that the deleted blog only returns a subset of fields that do not include the `capabilities` because it goes through this branch of code fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qcyhtva%2Scebqhpgvba%2Swfba%2Qraqcbvagf%2Spynff.jcpbz%2Qwfba%2Qncv%2Qtrg%2Qfvgr%2Qraqcbvag.cuc%3Se%3Q8n211436%23395-og, so it only return these fields fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qcyhtva%2Scebqhpgvba%2Swfba%2Qraqcbvagf%2Spynff.jcpbz%2Qwfba%2Qncv%2Qtrg%2Qfvgr%2Qraqcbvag.cuc%3Se%3Q8n211436%2396-og

I will stop investigating that as returning additional `capabilitites` could cause some regression.

> I think I am fine with the first approach. Testing it now.

Ok, I will restore the original commit [6f32b3d](https://github.com/Automattic/wp-calypso/pull/97547/commits/6f32b3d7f22bc877c0697f07f3e26e41429b902d) and add additional comments

Reply:
I had this commit https://github.com/Automattic/wp-calypso/pull/97547/commits/b00a74f72ebab50376277095e7c5e408fa4b79dd in mind, which relies only in the backend `manage_options`.

Reply:
Ok, got it, sorry for the confusion. I have added my comments here [e921c4a](https://github.com/Automattic/wp-calypso/pull/97547/commits/e921c4a650879c14629690ce9f25ec905798af55) so I guess this PR is ready to land tomorrow 😄